improver/utilities/mathematical_operations.py

```diff
@@ -200,68 +230,64 @@ def perform_integration(
+                          self.direction_of_integration,
+                          self.start_point, self.end_point):
+                continue
 
-        # Perform the integration
-        stride_sum = 0
-        integrated_cubelist = iris.cube.CubeList([])
-        for (upper_bounds_slice, lower_bounds_slice,
-             integrated_slice) in levels_tuple:
-            upper_bound = (
-                upper_bounds_slice.coord(
-                    self.coord_name_to_integrate).points.item())
-            lower_bound = (
-                lower_bounds_slice.coord(
-                    self.coord_name_to_integrate).points.item())
-            if not self.start_point and not self.end_point:
-                pass
-            elif self.start_point:
-                if self.direction_of_integration == "positive":
-                    if lower_bound < self.start_point:
-                        continue
-                elif self.direction_of_integration == "negative":
-                    if upper_bound > self.start_point:
-                        continue
-            elif self.end_point:
-                if self.direction_of_integration == "positive":
-                    if upper_bound > self.end_point:
-                        continue
-                elif self.direction_of_integration == "negative":
-                    if lower_bound < self.end_point:
-                        continue
             stride = np.abs(upper_bound - lower_bound)
-            upper_half_of_stride = upper_bounds_slice.data * 0.0
-            # Restrict the integration to only consider positive values.
-            # This condition is specific for the computation of the wet-bulb
-            # temperature integral.
-            uindex = np.where(upper_bounds_slice.data > 0)
-            upper_half_of_stride[uindex] = (upper_bounds_slice.data[uindex] *
-                                            0.5 * stride)
-            # Restrict the integration to only consider positive values.
-            # This condition is specific for the computation of the wet-bulb
-            # temperature integral.
-            lindex = np.where(lower_bounds_slice.data > 0)
-            lower_half_of_stride = lower_bounds_slice.data * 0.0
-            lower_half_of_stride[lindex] = (lower_bounds_slice.data[lindex] *
-                                            0.5 * stride)
-            stride_sum += lower_half_of_stride + upper_half_of_stride
-            integrated_slice.data = stride_sum
-            integrated_cubelist.append(integrated_slice.copy())
-
-        if len(integrated_cubelist) == 0:
+            upper_half_data = np.where(
+                upper_bounds_slice.data > 0,
+                upper_bounds_slice.data * 0.5 * stride, 0.0)
+            lower_half_data = np.where(
+                lower_bounds_slice.data > 0,
+                lower_bounds_slice.data * 0.5 * stride, 0.0)
+            stride_sum += upper_half_data + lower_half_data
```

Comment:
Can we rename this as `integral_data` or something? `stride_sum` confused me in a context that also contains `stride`.

Reply:
Done